components/logins/src/migrate_sqlcipher_db.rs

```diff
@@ -67,40 +99,95 @@ fn sqlcipher_3_compat(conn: &Connection) -> Result<()> {
     Ok(())
 }
 
-fn sqlcipher_export(conn: &mut Connection, new_db_path: impl AsRef<Path>) -> Result<()> {
-    // Export sqlite data to a plaintext DB using the strategy from example 2 here:
-    // https://www.zetetic.net/sqlcipher/sqlcipher-api/index.html#sqlcipher_export
-    let path = new_db_path.as_ref().as_os_str();
-    let path = path
-        .to_str()
-        .ok_or_else(|| ErrorKind::InvalidPath(path.to_os_string()))?;
-
-    conn.execute("ATTACH DATABASE ? AS plaintext key ''", &[path])?;
-    // this one is a bit weird because it's a SELECT statement that we know will return 0 rows.
-    // rusqlite will return an Error if we use execute(), so we use query_row with a dummy closure
-    conn.query_row("SELECT sqlcipher_export('plaintext')", NO_PARAMS, |_| Ok(0))?;
-    conn.execute("DETACH DATABASE plaintext", NO_PARAMS)?;
-    Ok(())
+//Manually copy over row by row from sqlcipher db to a plaintext db
+pub fn migrate_from_sqlcipher_db(
+    cipher_conn: &mut Connection,
+    new_db_store: LoginStore,
+    encryption_key: &str,
+) -> Result<MigrationMetrics> {
+    let start_time = Instant::now();
+
+    let mut metrics: MigrationMetrics = MigrationMetrics::default();
+
+    // Select From both LoginsL and LoginsM with a union to ensure we're covering our
+    // migration cases (one table has but not the other, etc)
+    let mut select_stmt = cipher_conn.prepare(&GET_BY_GUID_SQL)?;
+
+    // Use raw rows to avoid extra copying since we're looping over an entire table
+    let mut rows = select_stmt.query(NO_PARAMS)?;
+    while let Some(row) = rows.next()? {
```

Comment:
Maybe wrap everything inside the brace in a function so that we can handle errors and continue on.  For example if one of the required fields were missing or some field had the wrong type.